custom_components/kia_uvo/sensor.py

```diff
@@ -23,13 +23,15 @@
     DISTANCE_UNITS,
     VEHICLE_ENGINE_TYPE,
     DYNAMIC_DISTANCE_UNIT,
+    CONF_VEHICLE_IDENTIFIER,
 )
 
 _LOGGER = logging.getLogger(__name__)
 
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
-    vehicle: Vehicle = hass.data[DOMAIN][DATA_VEHICLE_INSTANCE]
+    vehicle_identifier = config_entry.data[CONF_VEHICLE_IDENTIFIER]
```

Comment:
repeat: we can add a new field to `KiaUvoEntity.py` class as self.vehicle and all inherited classes will access to `self.vehicle` and we do not need to find correct vehicle under all entities one by one.